specs/extensions/openapi-extensions.md

```diff
@@ -127,4 +129,29 @@ paths:
                 - "I **should not** respond with either sample or full form of lyrics to the song because the song created by createSong(**params) is shown to the user directly along with the lyrics."
                 - "If createSong(**params) was not invoked because the service was overloaded or the user reached their invocation limit, I **must** tell the user they can visit the https://app.suno.ai website to create more songs and display the link using Markdown syntax."
                 - "I **shall** tell the user they can visit the https://app.suno.ai website to create more songs and display the link using Markdown syntax."
+```
+
+## x-openai-isConsequential
```

Comment:
We should probably generate the confirmation dialog part of the plugin if isConsequential is set to true or not set at all for anything else than GET.

Reply:
I don't think so. There are 2 different aspects here:
1 - isConsequential is to tell whether we should prompt the confirmation dialog. This will be used by Copilot and Copilot will be responsible showing the confirmation dialog based on the rules informed in this doc. Kiota should not change anything and let Copilot handle it.
2 - confirmation object is something that the user can provide to customize how Copilot is going to prompt the confirmation to the user, like what is title, body and the type (like adaptive cards). This is something we (as Kiota generator) shouldn't be assuming what users want. With overlay, sure, we can provide an easier way for folks to add confirmation capability, but as for generation, I don't think we should be generating them on behalf of the user.

Reply:
Overlay? It's not going to end up in the OpenAPI, so not sure if it's the right place. I think we can infer a lot of it from the OpenAPI and generate the confirmation dialog!

Reply:
But if there is `isConsequential` true and there is no confirmation object provided, Copilot will infer what do. What is the benefit of us doing it in the generation?